src/ProductPage.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState, useEffect } from 'react';
+import css from './ProductPage.module.css';
+
+interface Product {
+  category: {
+    id: number,
+    name: string,
+    image: string
+  };
+  creationAt: string;
+  description: string;
+  id: number;
+  images: string[];
+  price: number;
+  title: string;
+  updatedAt:  string;
+}
+
+const ProductPage: React.FC = () => {
+  const [products, setProducts] = useState<Product[]>([]);
+  const [searchTerm, setSearchTerm] = useState<string>('');
+  const [sortBy, setSortBy] = useState<string>('');
+  const [filterOptions, setFilterOptions] = useState<{ new: boolean; old: boolean }>({
+    new: false,
+    old: false
+  });
+
+  useEffect(() => {
+    fetchProducts();
+  }, []);
+
+  const fetchProducts = async () => {
+    // Симуляція отримання даних з сервера
+    const response = await fetch('https://api.escuelajs.co/api/v1/products?limit=0&offset=0');
+    const data = await response.json();
+    console.log(data);
+    setProducts(data);
+  };
+
+  const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchTerm(event.target.value);
+    filterProducts();
+  };
+
+  const handleSortChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSortBy(event.target.value);
+    sortProducts();
+  };
+
+  const handleFilterChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, checked } = event.target;
+    setFilterOptions({ ...filterOptions, [name]: checked });
+    filterProducts();
+  };
+
+  const filterProducts = () => {
+    let filteredProducts = [...products];
+    if (filterOptions.new) {
+      filteredProducts = filteredProducts.filter((product) => product.creationAt > '2021-01-01');
```

Comment:
According to the task, the product is considered new if it has been created this year.
```suggestion
      filteredProducts = filteredProducts.filter((product) => new Date(product.creationAt).getFullYear() === new Date().getFullYear() );
```